app/controllers/saml_idp_controller.rb

```diff
@@ -8,15 +8,14 @@ class SamlIdpController < ApplicationController
   include SamlIdpLogoutConcern
   include FullyAuthenticatable
   include VerifyProfileConcern
+  include VerifySPAttributesConcern
 
   skip_before_action :verify_authenticity_token
 
   def auth
     return confirm_two_factor_authenticated(request_id) unless user_fully_authenticated?
-    process_fully_authenticated_user do |needs_idv, needs_profile_finish|
-      return redirect_to(verify_url) if needs_idv && !needs_profile_finish
-      return redirect_to(account_or_verify_profile_url) if needs_profile_finish
-    end
+    verify_action = build_url_for_fully_authenticated_user
+    return redirect_to(verify_action) if verify_action.present?
```

Comment:
Since we're making changes here, what do you think about having the same code in both OIDC and SAML to ensure consistency? The OIDC code is probably the one to use.